### ISSUE TITLE
Ability to modify font, font size and add custom colour schemes

### ISSUE THREAD

**Issue by joeljose350** [elementary/terminal#648]
### Problem

Even though the [pantheon-tweaks](https://github.com/pantheon-tweaks/pantheon-tweaks) app allows for more customizability, pantheon-terminal is still not as customisable as gnome-terminal and other popular terminal clients. Being able to add a custom colour scheme and changing the font and font size is a very basic feature and should be implemented in this app to make it as good as the other terminal clients.

### Proposal

Offering users the ability to modify the font and font size, add custom colour schemes and export their settings as a JSON file would be a great feature, in my opinion and make pantheon-terminal more attractive to new users.

### Prior Art (Optional)

![Screenshot from 2022-01-02 00-38-47](https://user-images.githubusercontent.com/53969297/147858269-be4f4f48-29db-48cd-9fda-0e29f8268a4a.png)

**Comment by cassidyjames** [comment]
You can currently modify the font size from the settings menu, and it persists across tabs/sessions. There is an open issue for color schemes at #418. Exporting as a JSON file is… interesting, but I don't _really_ see how that attracts new users as it seems fairly advanced, and you could relatively easily export and import GSettings from Terminal if you have custom settings—though perhaps an easier GUI way to do that could be proposed.

So I _think_ the main proposal here is for a graphical way to change the font? If that's the case, I think it would be worth considering this from a platform accessibility standpoint and enable changing the monospoace font from System Settings → Desktop → Text.

@joeljose350 can I ask the reasoning you have for changing the font? Is it for familiarity/aesthetic preference, legibility, or something else?

**Comment by joeljose350** [comment]
> @joeljose350 can I ask the reasoning you have for changing the font? Is it for familiarity/aesthetic preference, legibility, or something else?

I keep the font really large on text editors and the terminal to make the text easier to read. I don't really want to have huge fonts for any other app, so being able to set a different font for the terminal in a graphical way would be a good feature in my opinion.

**Comment by cassidyjames** [comment]
>being able to set a different font for the terminal in a graphical way would be a good feature 

Right, that makes sense and exists in the existing settings menu. :smile: It sets it for the current tab and will persist for new tabs unless/until changed again. It also matches the design in Code.

![Screenshot from 2022-01-03 10-25-05](https://user-images.githubusercontent.com/611168/147960570-e9b249d3-3f7e-4ca5-bb38-af44f13ccaad.png)

Does that not work for your use case?

**Comment by joeljose350** [comment]
I was thinking of using another font altogether, not the one used by the system. I really like elementary's default font, but I usually use Roboto Mono in the terminal and text editors. So if someone wanted to do something like this, having a way to do it graphically would be really helpful.

**Comment by cassidyjames** [comment]
@joeljose350 alright, so if the issue is changing the monospace font used, I agree that could be useful (again, thinking of accessibility in addition to aesthetic preference), but I do think that belongs in the System Settings so it affects Terminal, Code, and the rest of the system for any related components.

I understand you came asking for "a separate settings menu with sub-menus to customize the editor graphically", but we try to balance arbitrary "customization" that can make the software more confusing and harder to maintain with the underlying problems we're actually trying to solve. I think #418 covers the want for user-provided and configurable color schemes, so I'm going to move this to the appropriate System Settings pane to be able to set it globally.
